admin_ui_support/src/Routing/RouteSubscriber.php

```diff
@@ -11,10 +12,27 @@
  */
 class RouteSubscriber extends RouteSubscriberBase {
 
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * RouteSubscriber constructor.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->configFactory = $config_factory;
+  }
+
   /**
    * {@inheritdoc}
    */
   protected function alterRoutes(RouteCollection $collection) {
+    if (!$this->configFactory->get('admin_ui_support.settings')->get('redirect_related_routes')) {
```

Comment:
Did you thought about making this distinction on runtime, so you don't need to rebuild the router for that?

Reply:
Well we are already replacing the controller so we would have lost the original controller.

I don't think this is something that would happen very often at all. So once the routes are rebuilt there is not effect at runtime which seems better for something that would switch very infrequently.

Reply:
Yeah fair, let's not worry for now :)